src/commons.ts

```diff
@@ -131,7 +131,17 @@ export default class Commons {
         const tempObj = JSON.parse(customSettingStr);
 
         Object.assign(customSettings, tempObj);
-        customSettings.token = customSettings.token.trim();
+        if (customSettings.token) {
+          customSettings.token = customSettings.token.trim();
+        } else {
+          Commons.LogException(

```

Comment:
can you remove this. I suggest to keep the exception handle on the previous location. Although we can improve the exception message more better to see why its causing this behaviour

Reply:
Hum, I consider it is a good place for this.

Why do you prefer to keep the previous location ? :)

If we keep this place, should we reproduce the behaviour with putting `customSettings = null;` ?

Reply:
> If we keep this place, should we reproduce the behaviour with putting customSettings = null; ?

Yes you want to test your build on other users #1078 i would suggest to create your package of your build and upload and see the problems users are facing.

>Why do you prefer to keep the previous location ? :)

Exception can be arised from code while reading the file and other lines you can see. So i suggest to handle all the code which can raise exception in one place ( like old location) and what we can do is to improve the exception message if there is something we can do to improve.

Reply:
I think the previous exception is thrown because of the runtime exception of JS trying to `null.trim()`

Basically this code just ensures that we protect against that runtime exception, by raising the exception but not in a fatal way. Can you add code to re-prompt for the token instead of raising this exception?

Reply:
> I think the previous exception is thrown because of the runtime exception of JS trying to null.trim()

Exactly !

Can you remove the exception line you have written there and let the main `catch` handle this.

> Can you add code to re-prompt for the token instead of raising this exception?

I dont think it will raise this exception when `null.trim()` is fixed and it should return `token` as null to the main driver classes, they are responsible to do something for token. 

If the token is not found, itshould open the welcome page instead to allow the user to configure the Settings Sync.